From 2ba92da9fac9370c4b39caf66b6a94bebd859afc Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Wed, 22 Mar 2017 15:30:59 +0100
Subject: [PATCH] Don't search default paths when looking for numpy

cmakes find_path() function looks first in system directories and then
in the directories listed by PATHS.  So if there are numpy includes in
/usr/include they will be found first which can break the build.

One way to fix this could be to use HINTS instead of PATHS which will be
searched before the system paths. But in this case we know which
directory we want as we got it from python so let's just skip all other
paths and just look there by adding NO_DEFAULT_PATH.
---
 cmake/modules/FindNumPy.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/FindNumPy.cmake b/cmake/modules/FindNumPy.cmake
index 04a3a12..2a9bef8 100644
--- a/cmake/modules/FindNumPy.cmake
+++ b/cmake/modules/FindNumPy.cmake
@@ -82,7 +82,8 @@ if("${_VER_CHECK}" STREQUAL "")
     return()
 endif()
 
-find_path(NUMPY_INCLUDE_DIR numpy/numpyconfig.h PATHS ${_numpy_include_path})
+# we already know where it should be so don't search default paths as well.
+find_path(NUMPY_INCLUDE_DIR numpy/numpyconfig.h PATHS ${_numpy_include_path} NO_DEFAULT_PATH)
 if(NUMPY_INCLUDE_DIR)
   set(NUMPY_INCLUDE_DIRS ${NUMPY_INCLUDE_DIR})
 endif()
